### PR TITLE
api: isolate schemas in `osrd_schemas` package

### LINKED ISSUE #1150: Isolate schemas
## Short description meaningful for devs and users 

- [ ] Isolate and create a importable python module.
- [ ] Find how it can be imported by poetry (with/without a registry)
- [ ] Add the dependency to `api` and `infra-generator`

### PR DESCRIPTION
**Pull request by bloussou**
This PR is moving `schemas` in a separate python package.

- `api` code is now under `python/api`
- `api/osrd_infra/schemas/` -> `python/osrd_schemas`
- `python folder as been created because of docker build context, IMO it is the simplest way to have something working : https://github.com/docker/compose/issues/9461
- Added linter checks in ci for `osrd_schemas` package

close [#1150](https://github.com/DGEXSolutions/osrd/issues/1150)

### CODE REVIEW

File: core/examples/generated/lib/railjson_generator/__init__.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import sys
 from pathlib import Path
 
-api_schema_parent_path = Path(__file__).parents[5] / "api" / "osrd_infra" 
+api_schema_parent_path = Path(__file__).parents[5] / "python" / "osrd_schemas"
```
Comment (flomonster) [inline core/examples/generated/lib/railjson_generator/__init__.py:4]: Now that schemas are isolated. We should be able to `pip install --dev` it as a library. We can plan to do it in a separate PR though.

**Review by flomonster — CHANGES_REQUESTED** [review]
Thanks for this PR.

Few files should be updated:
- `.github/dependabot` 
- `.github/CODEOWNERS`

File: core/examples/generated/lib/railjson_generator/__init__.py (hunk 1 above)
Comment (bloussou) [reply core/examples/generated/lib/railjson_generator/__init__.py:4]: Yes it is what I thought :)

**Review by flomonster — APPROVED** [review]

**Review by multun — APPROVED** [review]
